ext/reflection/php_reflection.c

```diff
@@ -2690,6 +2754,116 @@ ZEND_METHOD(reflection_parameter, isVariadic)
+   Returns whether parameter MUST be an array */
+ZEND_METHOD(reflection_typehint, isArray)
+{
+	reflection_object *intern;
+	typehint_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->type_hint == IS_ARRAY);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionTypeHint::isCallable()
+   Returns whether parameter MUST be callable */
+ZEND_METHOD(reflection_typehint, isCallable)
+{
+	reflection_object *intern;
+	typehint_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->type_hint == IS_CALLABLE);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionTypeHint::isNullable()
+  Returns whether parameter MAY be null */
+ZEND_METHOD(reflection_typehint, isNullable)
+{
+	reflection_object *intern;
+	typehint_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->allow_null);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionTypeHint::isInstance()
+  Returns whether parameter MUST be a class/interface instance */
+ZEND_METHOD(reflection_typehint, isInstance)
+{
+	reflection_object *intern;
+	typehint_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->type_hint == IS_OBJECT);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionTypeHint::isScalar()
+  Returns whether parameter MUST be scalar */
+ZEND_METHOD(reflection_typehint, isScalar)
+{
+	reflection_object *intern;
+	typehint_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->type_hint == IS_STRING ||
+		param->arg_info->type_hint == _IS_BOOL ||
+		param->arg_info->type_hint == IS_LONG ||
+		param->arg_info->type_hint == IS_DOUBLE);
+}
+/* }}} */
+
+/* {{{ proto public string ReflectionTypeHint::__toString()
+   Return the text of the type hint */
+ZEND_METHOD(reflection_typehint, __toString)
+{
+	reflection_object *intern;
+	typehint_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	switch (param->arg_info->type_hint) {
+		case IS_ARRAY:    RETURN_STRINGL("array", sizeof("array") - 1);
+		case IS_CALLABLE: RETURN_STRINGL("callable", sizeof("callable") - 1);
+		case IS_OBJECT:   RETURN_STR(zend_string_copy(param->arg_info->class_name));
+		case IS_STRING:   RETURN_STRINGL("string", sizeof("string") - 1);
+		case _IS_BOOL:    RETURN_STRINGL("bool", sizeof("bool") - 1);
+		case IS_LONG:     RETURN_STRINGL("int", sizeof("int") - 1);
+		case IS_DOUBLE:   RETURN_STRINGL("float", sizeof("float") - 1);
+		default:
+			php_error_docref(NULL, E_ERROR, "Unknown type hint: %d", (int)param->arg_info->type_hint);
+			RETURN_EMPTY_STRING();
```

Comment:
Should use `EMPTY_SWITCH_DEFAULT_CASE()` here instead of the `E_ERROR` to assert that the switch enumerates all options.